src/Chainweb/Pact/PactService.hs

```diff
@@ -616,18 +615,37 @@ readAccountGuard pdb account
 
 -- | Calculate miner reward. We want this to error hard in the case where
 -- block times have finally exceeded the 120-year range. Rewards are calculated
--- in 500k steps
+-- at regular blockheight intervals.
+--
+-- See: 'rewards/miner_rewards.csv'
 --
 minerReward
     :: forall cas
     . BlockHeight -> PactServiceM cas P.ParsedDecimal
-minerReward bh = do
-    m <- view $ psMinerRewards . at (roundBy bh 500000)
-    case m of
+minerReward bh = use psMinerRewards >>= go
+  where
+    go (MinerRewards rewards q) = do
+      when (V.null q) err
+      let (!h, !t) = (V.unsafeHead q, V.unsafeTail t)
+      if bh <= h
+      then lookup_ h rewards
+      else modifyRewards t rewards
```

Comment:
The rewards map is now part of pact service state, maintaining a list of sorted keys. This list acts as a kind of queue, where each block height is compared against head and looked up at the appropriate upper bound in the map. There is no closed form for this. If the blockheight exceeds `head`'s value, then `head` is popped off the list and the state is updated with `tail`.